src/ucp/core/ucp_am.c

```diff
@@ -892,9 +894,10 @@ ucp_am_try_send_short(ucp_ep_h ep, uint16_t id, uint32_t flags,
                       const void *buffer, size_t length)
 {
     if (ucs_unlikely(((length != 0) && (header_length != 0)) ||
-                     ((ssize_t)(length + header_length) >
-                      ucp_ep_config(ep)->am_u.max_eager_short)) ||
-                     (flags & UCP_AM_SEND_FLAG_RNDV)) {
+                     ucp_proto_eager_is_inline(ep,
+                                            &ucp_ep_config(ep)->am_u.max_eager_short,
```

Comment:
unaligned?

Reply:
fitting into 85 lines

Reply:
hard to read.. can you pls break it to several if with goto's?